pkg/hubble/filters/tcp.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020 Authors of Hubble
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filters
+
+import (
+	"context"
+	"fmt"
+
+	flowpb "github.com/cilium/cilium/api/v1/flow"
+	v1 "github.com/cilium/cilium/pkg/hubble/api/v1"
+)
+
+func filterByTCPFlags(flags []*flowpb.TCPFlags) (FilterFunc, error) {
+	return func(ev *v1.Event) bool {
+		l4tcp := ev.GetFlow().GetL4().GetTCP()
```

Comment:
maybe just some safe guard here, similar to https://github.com/cilium/cilium/blob/master/pkg/hubble/api/v1/flow.go#L39

Reply:
For what it's worth, the `GetL4()` and `GetTCP()` (as all protobuf getters) both support `nil` receivers. So the current code should be nil-pointer safe:

https://github.com/cilium/cilium/blob/0ea1b88b52a08d72d23314b924fea911140ff3db/api/v1/flow/flow.pb.go#L988-L1000

Did you have additional safe-guards in mind, @sayboras?

Reply:
I think it is more than enough, thanks :100: